src/mongocrypt-ctx.c

```diff
@@ -587,41 +577,61 @@ mongocrypt_ctx_setopt_masterkey_aws (mongocrypt_ctx_t *ctx,
       return false;
    }
 
-   if (ctx->opts.masterkey_kms_provider) {
+   if (ctx->opts.kek.kms_provider != MONGOCRYPT_KMS_PROVIDER_AWS &&
+       ctx->opts.kek.kms_provider != MONGOCRYPT_KMS_PROVIDER_NONE) {
+      return _mongocrypt_ctx_fail_w_msg (ctx, "master key already set");
```

Comment:
I tried to limit the amount of changed error messages for now. This continues to have the "master key" terminology.